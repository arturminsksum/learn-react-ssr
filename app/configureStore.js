import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';

import articlesApp from './reducers';

export default initialState => {
  const store = createStore(
    articlesApp,
    initialState,
    composeWithDevTools(applyMiddleware(thunk)),
  );
  return store;
};
