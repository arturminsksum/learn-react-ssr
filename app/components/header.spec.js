import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import Header from './header';

const mockStore = configureStore();

describe('Header', () => {
  it('should render Header correctly for unlogged user', () => {
    const store = mockStore({ logged: false });
    const wrapper = mount(
      <BrowserRouter>
        <Provider store={store}>
          <Header />
        </Provider>
      </BrowserRouter>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Header correctly for logged user', () => {
    const store = mockStore({ logged: true });
    const wrapper = mount(
      <BrowserRouter>
        <Provider store={store}>
          <Header />
        </Provider>
      </BrowserRouter>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
