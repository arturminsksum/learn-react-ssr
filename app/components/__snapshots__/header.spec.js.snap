// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render Header correctly for logged user 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "blank",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Connect(Header)>
        <Header
          handleLogout={[Function]}
          logged={true}
        >
          <nav
            aria-label="main navigation"
            className="navbar"
            role="navigation"
          >
            <div
              className="navbar-brand"
            >
              <Link
                className="navbar-item title"
                replace={false}
                to="/"
              >
                <a
                  className="navbar-item title"
                  href="/"
                  onClick={[Function]}
                >
                  ArchiNews
                </a>
              </Link>
            </div>
            <div
              className="navbar-menu"
            >
              <Link
                className="navbar-item"
                replace={false}
                to="/"
              >
                <a
                  className="navbar-item"
                  href="/"
                  onClick={[Function]}
                >
                  Home
                </a>
              </Link>
              <Link
                className="navbar-item"
                replace={false}
                to="/add"
              >
                <a
                  className="navbar-item"
                  href="/add"
                  onClick={[Function]}
                >
                  Add Post
                </a>
              </Link>
            </div>
            <div
              className="navbar-end"
            >
              <div
                className="navbar-item"
              >
                <button
                  className=" button is-light"
                  onClick={[Function]}
                >
                  LogOut
                </button>
              </div>
            </div>
          </nav>
          <section
            className="hero is-primary"
          >
            <div
              className="hero-body"
            >
              <div
                className="container"
              >
                <h1
                  className="title"
                >
                  Choose your channel
                </h1>
                <h2
                  className="subtitle"
                >
                  All sources with English news
                </h2>
              </div>
            </div>
          </section>
        </Header>
      </Connect(Header)>
    </Provider>
  </Router>
</BrowserRouter>
`;

exports[`Header should render Header correctly for unlogged user 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "blank",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Connect(Header)>
        <Header
          handleLogout={[Function]}
          logged={false}
        >
          <nav
            aria-label="main navigation"
            className="navbar"
            role="navigation"
          >
            <div
              className="navbar-brand"
            >
              <Link
                className="navbar-item title"
                replace={false}
                to="/"
              >
                <a
                  className="navbar-item title"
                  href="/"
                  onClick={[Function]}
                >
                  ArchiNews
                </a>
              </Link>
            </div>
            <div
              className="navbar-menu"
            >
              <Link
                className="navbar-item"
                replace={false}
                to="/"
              >
                <a
                  className="navbar-item"
                  href="/"
                  onClick={[Function]}
                >
                  Home
                </a>
              </Link>
              <Link
                className="navbar-item"
                replace={false}
                to="/add"
              >
                <a
                  className="navbar-item"
                  href="/add"
                  onClick={[Function]}
                >
                  Add Post
                </a>
              </Link>
              <Link
                className="navbar-item"
                replace={false}
                to="/login"
              >
                <a
                  className="navbar-item"
                  href="/login"
                  onClick={[Function]}
                >
                  LogIn
                </a>
              </Link>
              <Link
                className="navbar-item"
                replace={false}
                to="/signup"
              >
                <a
                  className="navbar-item"
                  href="/signup"
                  onClick={[Function]}
                >
                  SignUp
                </a>
              </Link>
            </div>
          </nav>
          <section
            className="hero is-primary"
          >
            <div
              className="hero-body"
            >
              <div
                className="container"
              >
                <h1
                  className="title"
                >
                  Choose your channel
                </h1>
                <h2
                  className="subtitle"
                >
                  All sources with English news
                </h2>
              </div>
            </div>
          </section>
        </Header>
      </Connect(Header)>
    </Provider>
  </Router>
</BrowserRouter>
`;
